app/services/status_update_service.rb

```diff
@@ -7,7 +7,7 @@ def initialize(reference:, status:, user: nil)
   end
 
   def self.call(*args)
-    new(*args).call
+    DeliveryAttempt.transaction { new(*args).call }
```

Comment:
If we put the transaction here, I think we should put in a `private_class_method :new` to avoid anyone accidentally doing `.new(...).call`.

Reply:
Think there is already one.

Reply:
Oh so there is... oops